src/FlowPlayer.cpp

```diff
@@ -75,6 +75,11 @@ static void migrateCache()
         }
         QDir(QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation)).rmdir("flowplayer");
     }
+    // if the media-art directory does not exist, make it.
```

Comment:
This comment is consistent with point 2 in the PR description, but AFAICS not with the code: Line 79 checks if the `newCache` directory exists, not its subdirectory `media-art`, right?

Reply:
I added the mkdir at the end since on a fresh install, the needed directories were not being added. But, this is a bit hacky. My feeling was that:
1. The migration code should go away (since it's trivial to rebuild cache data)
2. Sailjail should be on be deaulft (after which the migration code will fail)
3. The directory creation code should go elsewhere. It ended up there while I was investigating why the 'media-art' directory was not being created.